### PR TITLE
Teach drag start behaviors to DragGestureRecognizer final

### PR DESCRIPTION
**Pull request by jslavitz**
Just adds a couple fixes to #18314

### CODE REVIEW

**Comment by Hixie** [comment]
cc @goderbauer

**Comment by goderbauer** [comment]
This seems to break a couple of tests on cirrus. @jslavitz Can you take a look at those?

**Comment by Hixie** [comment]
@jslavitz and I studied this and concluded that we'll need to expose a way to control this flag from ListView, GridView, Dismissible, and some other widgets.

**Comment by Hixie** [comment]
(and then most of the tests can just set that flag and don't need to be updated.)

File: packages/flutter/lib/src/gestures/monodrag.dart [hunk 1]
```diff
@@ -48,7 +59,14 @@ typedef GestureDragCancelCallback = void Function();
...
   /// Initialize the object.
-  DragGestureRecognizer({ Object debugOwner }) : super(debugOwner: debugOwner);
+  DragGestureRecognizer({
+    @required this.dragStartBehavior,
+    Object debugOwner,
+  })  : assert(dragStartBehavior != null),
+        super(debugOwner: debugOwner);
+
+  /// Configure the behavior of offsets sent to [onStart].
+  final DragStartBehavior dragStartBehavior;
```
Comment (Hixie) [inline packages/flutter/lib/src/gestures/monodrag.dart]: this will need to be mutable so that the initializers in GestureDetector can update the field.

**Pull request closed without merging**